R/tidyverse.R

```diff
@@ -401,3 +404,28 @@ base_and_recommended <- function() {
     "survival", "tcltk", "tools", "utils"
   )
 }
+
+#' @rdname tidyverse
+#' @inheritParams use_logo
+#' @export
+use_tidy_logo <- function(geometry = "240x278", retina = TRUE) {
+  if (!is_rstudio_pkg()) {
+    ui_stop("This function can only be used for Posit packages")
+  }
+
+  pkg <- project_name()
+  owner <- "rstudio"
+  repo <- "hex-stickers"
+
+  png_contents <- gh::gh("GET /repos/{owner}/{repo}/contents/PNG/",
+                         owner = owner, repo = repo)
+
+  pkg_png <- purrr::keep(png_contents, ~ .x[["name"]] == paste0(pkg, ".png"))
+
+  if (length(pkg_png) == 0L) {
+    ui_stop("No png logo available for {pkg} at https://github.com/{owner}/{repo}/")
+  }
+
+  url <- pkg_png[[1]]$download_url
+  use_logo(url, geometry = geometry, retina = retina)
```

Comment:
What do you think of just directly grabbing the file? E.g.,

``` r
tf <- withr::local_tempfile(fileext = ".png")
gh::gh(
  "/repos/rstudio/hex-stickers/contents/PNG/{pkg}.png/",
  pkg = project_name(),
  .destfile = tf,
  .accept = "application/vnd.github.v3.raw"
)
use_logo(tf)
```

Reply:
I like that. It's the accept heard that I was missing! Do you think then just let it 404 and error that way if the package doesn't have a sticker? Or should I catch it and throw something more informative?

Reply:
gh does a pretty nice job of exposing lots of info upon failure, so I would just let its error bubble up if the file's not there.